pkg/topolvm-scheduler/cmd/root.go

```diff
@@ -80,23 +86,29 @@ func subMain() error {
 		return err
 	}
 
-	serv := &well.HTTPServer{
-		Server: &http.Server{
-			Addr:    config.ListenAddr,
-			Handler: h,
-		},
+	serv := &http.Server{
+		Addr:        config.ListenAddr,
+		Handler:     accessLogHandler(ctx, h),
+		ReadTimeout: 30 * time.Second,
 	}
 
-	err = serv.ListenAndServe()
-	if err != nil {
-		return err
-	}
-	err = well.Wait()
+	idleConnsClosed := make(chan struct{})
+	go func() {
```

Comment:
This goroutine will continue to run although ListenAndServe() returns an error. It should be made to stop correctly.

Reply:
I fixed my code. Please review this again. The change is the following: https://github.com/topolvm/topolvm/compare/ae2be29c5ce1f1ac5ab4fd6777a3532f4866154f..cc0be51f79191ebf06ecdbfd8c805013f846b13b